File: citaria_backend/src/main/java/com/citaria/modelo/Empleado.java
package com.citaria.modelo;

import jakarta.persistence.*;
import jakarta.validation.constraints.*;

import java.time.LocalDateTime;

/**
 * Empleado de una organización.
 * Incluye campo anonimizado_at para cumplimiento RGPD.
 */
@Entity
@Table(name = "empleado")
public class Empleado {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    @NotNull
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "organizacion_id", nullable = false)
    private Organizacion organizacion;

    @NotBlank
    @Size(max = 100)
    @Column(nullable = false, length = 100)
    private String nombre;

    @NotBlank
    @Size(max = 150)
    @Column(nullable = false, length = 150)
    private String apellidos;

    @Email
    @Size(max = 255)
    @Column(length = 255)
    private String email;

    @Size(max = 20)
    @Column(length = 20)
    private String telefono;

    @Size(max = 500)
    @Column(name = "foto_url", length = 500)
    private String fotoUrl;

    @NotNull
    @Column(nullable = false)
    private Boolean activo = true;

    @Column(name = "anonimizado_at")
    private LocalDateTime anonimizadoAt;

    public Integer getId() {
        return id;
    }

    public void setId(Integer id) {
        this.id = id;
    }

    public Organizacion getOrganizacion() {
        return organizacion;
    }

    public void setOrganizacion(Organizacion organizacion) {
        this.organizacion = organizacion;
    }

    public String getNombre() {
        return nombre;
    }

    public void setNombre(String nombre) {
        this.nombre = nombre;
    }

    public String getApellidos() {
        return apellidos;
    }

    public void setApellidos(String apellidos) {
        this.apellidos = apellidos;
    }

    public String getEmail() {
        return email;
    }

    public void setEmail(String email) {
        this.email = email;
    }

    public String getTelefono() {
        return telefono;
    }

    public void setTelefono(String telefono) {
        this.telefono = telefono;
    }

    public String getFotoUrl() {
        return fotoUrl;
    }

    public void setFotoUrl(String fotoUrl) {
        this.fotoUrl = fotoUrl;
    }

    public Boolean getActivo() {
        return activo;
    }

    public void setActivo(Boolean activo) {
        this.activo = activo;
    }

    public LocalDateTime getAnonimizadoAt() {
        return anonimizadoAt;
    }

    public void setAnonimizadoAt(LocalDateTime anonimizadoAt) {
        this.anonimizadoAt = anonimizadoAt;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (!(o instanceof Empleado that)) return false;
        return id != null && id.equals(that.id);
    }

    @Override
    public int hashCode() {
        return getClass().hashCode();
    }

    @Override
    public String toString() {
        return "Empleado{" +
                "id=" + id +
                ", organizacionId=" + (organizacion != null ? organizacion.getId() : null) +
                ", nombre='" + nombre + '\'' +
                ", apellidos='" + apellidos + '\'' +
                ", email='" + email + '\'' +
                ", activo=" + activo +
                ", anonimizadoAt=" + anonimizadoAt +
                '}';
    }
}

File: citaria_frontend/lib/data/models/empleado.dart
class Empleado {
  final int? id;
  final int? organizacionId;
  final String nombre;
  final String apellidos;
  final String? email;
  final String? telefono;
  final String? fotoUrl;
  final bool? activo;
  final DateTime? anonimizadoAt;

  const Empleado({
    this.id,
    this.organizacionId,
    required this.nombre,
    required this.apellidos,
    this.email,
    this.telefono,
    this.fotoUrl,
    this.activo,
    this.anonimizadoAt,
  });

  factory Empleado.fromJson(Map<String, dynamic> json) {
    return Empleado(
      id: json['id'] == null ? null : (json['id'] as num).toInt(),
      organizacionId: json['organizacionId'] == null
          ? null
          : (json['organizacionId'] as num).toInt(),
      nombre: json['nombre'] as String,
      apellidos: json['apellidos'] as String,
      email: json['email'] as String?,
      telefono: json['telefono'] as String?,
      fotoUrl: json['fotoUrl'] as String?,
      activo: json['activo'] as bool?,
      anonimizadoAt: json['anonimizadoAt'] == null
          ? null
          : DateTime.parse(json['anonimizadoAt'] as String),
    );
  }

  Map<String, dynamic> toJson() {
    return <String, dynamic>{
      'nombre': nombre,
      'apellidos': apellidos,
      'email': email,
      'telefono': telefono,
      'fotoUrl': fotoUrl,
      'activo': activo,
    };
  }
}
